### ISSUE TITLE
Mismatched Labels and Extracted Frames

### ISSUE THREAD

**Issue by Deep-Dark-Fantasyy** [DeepLabCut/DeepLabCut#2628]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Bug description

After I extracted frames, I began to label the new frames.
But after I labeled 10 videos, as I tried to toggle other layers except CollectedData, the napari window suddenly froze, then DLC window was dead and I saw errors reported in anaconda (sry, I forgot to take a screenshot of them). So I restarted anaconda and re-opened napari to continue to label other videos.

However, I find that all "unlabeled" videos' labels mismatch the frames. For example, I see labels (from previous correct CollectedData) for the 2nd frame showed up in the 1st frame. The orders are all wrong! I find it not a random mismatch, but caused by a few frames' delay.
Also, for those labeled videos, I checked that all labels match the frame order, but labels previously exist in machinelabel-iter (0~2) are gone! I could now only see labels in the CollectedData.

I restarted the anaconda, and even restarted my laptop, but neither helped. So I have questions below:
1. Is this mismatch simply a display error? Or the real data are truly wrong?
2. Is there a way to change the matching of labels and frames? Or the only solution is to re-label all the mismatched frames?

The examples below show that the labels in the 0th frame image are actually the labels of the 3rd frame image.  
<img width="779" alt="微信图片_20240619213556" src="https://github.com/DeepLabCut/DeepLabCut/assets/110468931/7da6ebe0-477b-4adb-8d50-6eb7868b9ae2">
<img width="779" alt="微信图片_20240619213601" src="https://github.com/DeepLabCut/DeepLabCut/assets/110468931/6e0456e9-004a-43e1-8348-8ba17b312ad8">

------------------------------------
### PROBLEM SOLVED!!!
I uninstalled and re-installed deeplabcut. At first, when I opened napari, I found that the labels were still mismatched. 
Then I noticed that on the top-right corner of deeplabcut I could choose "Engine".
I changed engine from "pytorch" to "tensorflow", and to my surprise, the labels went back to normal!
------------------------------------

### Operating System

Windows 11 (13th Gen Intel(R) Core(TM) i9-13900HX   2.20 GHz)

### DeepLabCut version

2.3.9

### DeepLabCut mode

single animal

### Device type

NIVAIA GeForce RTX 4060 Laptop GPU

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/DeepLabCut/DeepLabCut/blob/master/CODE_OF_CONDUCT.md)

**Comment by jeylau** [comment]
Hi @Deep-Dark-Fantasyy, thanks for reporting this issue. For the machinelabels files, this is the expected behavior: when hitting save, refined machine labels get automatically merged in the CollectedData file (https://github.com/DeepLabCut/napari-deeplabcut/tree/main?tab=readme-ov-file#save-layers).
As for the mismatch, I remember a similar issue that had been fixed [about a year ago](https://github.com/DeepLabCut/napari-deeplabcut/commit/23bbb6fe80c01c9a66a168f6db8e37a85865ae0b). Could you tell me what version of `napari-deeplabcut` you're running? It's visible under the Plugins > Install/Uninstall Plugins... menu when the labeling GUI is open.

**Comment by yO-Ole** [comment]
my napari-deeplabcut version is 0.2.1.6 @jeylau

**Comment by lpereiras17** [comment]
Hello,

I just had the same problem! It was labeling my bodyparts wrong and all over the place after I saved the data and opened the labeled folders again. Had the same deeplabcut-napari version and when I opened the Uninstall/Install to see the version a update button was shown. If you click on update the labels should be fixed!! The new version should be 0.2.1.7.

**Comment by jeylau** [comment]
Thanks @lpereiras17, that is what I was going to suggest. Run `pip install -U napari-deeplabcut` to make sure you're running the latest version.

**Comment by yO-Ole** [comment]
Thanks so much for your advice! @jeylau @lpereiras17 
I tried to update napari in anaconda, but still nothing changed.
So I uninstalled and re-installed deeplabcut. At first, when I opened napari, I found that the labels were still mismatched. Then I noticed that on the top-right corner of deeplabcut I could choose "Engine".
I changed engine from "pytorch" to "tensorflow", and to my surprise, the labels went back to normal!
